init.go

```diff
@@ -79,51 +75,25 @@ func Init(root, dir string, force bool) error {
 			dir, parentStack.PrjAbsPath())
 	}
 
-	logger.Trace().Msg("Get stack file.")
-
-	isInitialized := false
-
-	logger = log.With().
-		Str("action", "Init()").
-		Str("stack", dir).
-		Logger()
-
 	logger.Trace().Msg("Get stack info.")
 
 	parsedCfg, err := hcl.ParseDir(dir)
 	if err != nil {
 		return fmt.Errorf("checking config for stack %q: %w", dir, err)
 	}
 
-	isInitialized = parsedCfg.Stack != nil
-
-	if isInitialized && !force {
-		logger.Trace().Msg("Stack is initialized and not forced.")
-
-		logger.Trace().Msg("Checking version.")
-
-		if parsedCfg.Terramate == nil {
-			return fmt.Errorf("stack %q configuration has no 'terramate.required_version'", dir)
-		}
-
-		vconstraint := parsedCfg.Terramate.RequiredVersion
-
-		logger.Trace().Msg("Create new constraint from version.")
-
-		constraint, err := hclversion.NewConstraint(vconstraint)
-		if err != nil {
-			return fmt.Errorf("unable to check stack constraint: %w", err)
-		}
+	if parsedCfg.Stack != nil {
+		logger.Debug().Msg("Stack already initialized, nothing to do")
+		return nil
+	}
 
-		if !constraint.Check(tfversionObj) {
-			return fmt.Errorf("stack version constraint %q do not match terramate "+
-				"version %q", vconstraint, Version())
-		}
+	if !parsedCfg.IsEmpty() {
+		return errors.New("dir has terramate config with no stack defined, expected empty or a stack")
 	}
 
 	logger.Debug().Msg("Create new configuration.")
 
-	cfg, err := hcl.NewConfig(dir, DefaultVersionConstraint())
+	cfg, err := hcl.NewConfig(dir)
```

Comment:
required_version is only defined on root remember ? doesn't make sense to individual stack creation to affect root configuration IMHO (+ version checking is not required anymore either, if no version is defined on a project it means no version checking and it is a valid configuration).

More details: https://github.com/mineiros-io/terramate/blob/main/docs/project-config.md#required-version

Reply:
+ I also mentioned this on the PR comment: "Initializing stacks doesnt create any required version per stack anymore or do any version check". I think it makes sense :thinking:

Reply:
you're right but the docs dont make this clear.
What if someone puts a terramate.required_version in a stack file? does it just gets ignored?

Reply:
it vaguely says `if no terramate.required_version is defined on a project ...`.
I think we should make it clear that it can only be defined at the root directory.

Reply:
The config is inside project-config docs:

https://github.com/mineiros-io/terramate/blob/main/docs/project-config.md#required-version

At the beggining we of the "project config docs" we state:

"There can be only one Terramate project configuration on the entire project and it must be located at the project root dir."

The idea is that all configs documented as "Project Configuration" follow this behavior, or else I will repeate that for all configs documented there, would that be the idea ?

Reply:
The overall docs need fixes because it does talk about the terramate.config block as being the only project wide thing, which is not even true since terramate.config.generate is not a project wide/root only config either. Will try to improve.

Reply:
No, it's fine. I missed that part.

Reply:
> What if someone puts a terramate.required_version in a stack file? does it just gets ignored?

what about this?

Reply:
right now it is ignored, I will add an explicit check/error.

Reply:
as we discussed, checking this will be harder since there is no central load of all the configs for the whole project today, so getting consistent behavior won't be easy. Will be addressed later.